src/main/groovy/edu/oregonstate/mist/classsearchapi/resources/ClassSearchResource.groovy

```diff
@@ -0,0 +1,104 @@
+package edu.oregonstate.mist.classsearchapi.resources
+
+import com.codahale.metrics.annotation.Timed
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.AuthenticatedUser
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import edu.oregonstate.mist.classsearchapi.dao.ClassSearchDAO
+import io.dropwizard.auth.Auth
+
+import javax.validation.constraints.NotNull
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.Response.ResponseBuilder
+import javax.ws.rs.core.MediaType
+
+/**
+ * Sample resource class.
+ */
+@Path('/class-search/')
+class ClassSearchResource extends Resource {
+
+    public static final int TERM_LENGTH = 6
+    private final ClassSearchDAO classSearchDAO
+
+    ClassSearchResource(ClassSearchDAO classSearchDAO) {
+        this.classSearchDAO = classSearchDAO
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Timed
+    public Response classSearch(@Auth AuthenticatedUser _, @NotNull @QueryParam('term') String term,
+                               @QueryParam('subject') String subject,
+                               @QueryParam('courseNumber') String courseNumber, @QueryParam('q') String q) {
+        try {
+            // validate parameters
+            if (!term || !term.trim()) {
+                return badRequest("term is a required parameter.").build()
+            }
+
+            if (!term.isNumber() || term.length() != TERM_LENGTH) {
+                return badRequest("term should be a ${TERM_LENGTH} digit code").build()
+            }
+
+            if (!subject || !subject.trim()) {
+                return badRequest("subject is a required parameter").build()
+            }
+
+            def response = classSearchDAO.getData(term, subject, courseNumber, q, getPageNumber(), getPageSize())
+            //@todo: these params are calcualted twice :(
+            ResultObject resultObject = new ResultObject(data: response.data)
+            setPaginationLinks(response.sourcePagination, term, subject, courseNumber, q, resultObject)
+
+            ResponseBuilder responseBuilder = ok(resultObject)
+            responseBuilder.build()
+        } catch (Exception e) {
+            internalServerError("Woot you found a bug for us to fix!").build()
+        }
+    }
+
+    private void setPaginationLinks(def sourcePagination, String term, String subject, String courseNumber, String q,
+                                    ResultObject resultObject) {
```

Comment:
It's a little odd to have resultObject be an "out" parameter. You only set properties on resultObject.links. Can you just return a map?